demo/json_demo.py

```diff
@@ -0,0 +1,52 @@
+"""This file provides preliminary methods of accessing and storing JSON data."""
+import json
+import os
+
+
+def main():
+    """Pull data from a provided json, and write it back to the json."""
+    user_data_dict = get_dict_from_json_file("demofile")
+    print("Dictionary:\n")
+    for username in user_data_dict:
+        print(username + ":", user_data_dict[username], "\n")
+    json_format_string = write_dict_to_json_file(user_data_dict, "demofile")
+    print("Json-formatted string:", "\n", json_format_string, "\n")
+    new_user = {"testuser": {"commits": [], "issues": [], "pull_requests": []}}
+    json_format_string = add_user_to_json(
+        user_data_dict, new_user, "testuser", "demofile"
+    )
+    print("Json-formatted string:", "\n", json_format_string, "\n")
+
+
+def get_dict_from_json_file(json_file_name):
+    """Populate and return a dictionary of all the data in a specified json file."""
+    with open(os.path.join("./data/", json_file_name + ".json"), "r") as json_file:
+        # In the open() function, "r" specifies read-only access
+        json_string = json_file.read()
+        user_data_dict = json.loads(json_string)
+        # json.loads() converts a json-formatted string into a python dictionary
+    return user_data_dict
+
+
+def write_dict_to_json_file(user_data_dict, json_file_name):
+    """Overwrite specified json file with data from a given dictionary."""
+    with open(os.path.join("./data/", json_file_name + ".json"), "w") as json_file:
```

Comment:
The main() function of this file is primarily for testing (before I added test cases). My original intent for this file was to provide a callable list of functions more than a self-contained runnable program. Is a global variable for the path still a better idea through that framework?

Reply:
Yes I would think so. And by saying the path, I mean the `./data/` and the file extension `.json`, since they are somewhat hardcoded. Instead of calling `./data/` twice(or more), use a global variable somewhere that stores `./data/` and call the variable instead. That way, if next time any developer decide to change the directory name from `./data/` to something else, they don't have to go through each function and change each path by just modifying the global variable. Also, the global variable does not have to be in this one file. If you thing any of the other modules might also need to access to this path, then maybe it's a good idea just to create a module solely for saving these hardcoded values.